### PR TITLE
Fix search limit config item

### PR DESCRIPTION
**Pull request by mindset-tk**
The SQL query for the filterTimestamps function was malformed and would return the same (too high, low) timestamp every time, since it didn't filter for torrent name, and because max and min were reversed. It would never filter any torrent unless every single torrent had an extant entry in the db for every indexer.

Fixes cross-seed/cross-seed#453

### CODE REVIEW

**Comment by mindset-tk** [comment]
Hey, one other thing, I found that leaving line 85 on prefilter.ts as "min" will allow the tool to revisit indexers that were previously filtered by timestamps, see cross-seed/cross-seed/#402 - the code for performing the search is already sensitive to if an indexer was recently searched and will filter at that point.

File: src/preFilter.ts [hunk 1]
```diff
@@ -83,7 +86,7 @@ export async function filterTimestamps(searchee: Searchee): Promise<boolean> {
 			last_searched_all: db.raw("coalesce(timestamp.last_searched, 0)"),
 		})
 		.first();
-
+
```
Comment (bakerboy448) [inline src/preFilter.ts:89]: Erroneous white space?

Comment (mmgoodnow) [inline src/preFilter.ts:89]:
```suggestion

```

File: src/preFilter.ts [hunk 2]
```diff
@@ -83,7 +86,6 @@ export async function filterTimestamps(searchee: Searchee): Promise<boolean> {
 			last_searched_all: db.raw("coalesce(timestamp.last_searched, 0)"),
 		})
 		.first();
```
Comment (mmgoodnow) [inline src/preFilter.ts:88]:
```suggestion
		.first();

```

File: src/preFilter.ts (hunk 1 above)
Comment (mindset-tk) [reply src/preFilter.ts]: yes, erroneous.

**Comment by mmgoodnow** [comment]
are you really sure that `max` should be `min`?

File: src/preFilter.ts [hunk 3]
```diff
@@ -70,11 +70,14 @@ export async function filterTimestamps(searchee: Searchee): Promise<boolean> {
 			"timestamp.indexer_id": "indexer.id",
 			"timestamp.searchee_id": "searchee.id",
 		})
+		.where("searchee.name",
+		searchee.name
+		)
```
Comment (mmgoodnow) [inline src/preFilter.ts:75]: wow this one is embarrassing

File: src/preFilter.ts [hunk 4]
```diff
@@ -70,11 +70,14 @@ export async function filterTimestamps(searchee: Searchee): Promise<boolean> {
...
 		.whereIn(
 			"indexer.id",
 			enabledIndexers.map((i) => i.id)
 		)
-		.max({
+		.min({
```
Comment (mmgoodnow) [inline src/preFilter.ts:80]: I don't think this change is right. it would probably make more sense called `first_searched_any` though…

Comment (mmgoodnow) [reply src/preFilter.ts:80]: nvm I think I'm just getting confused

Comment (mmgoodnow) [reply src/preFilter.ts:80]: ok yeah I think I understand it now. we really want first_searched_any and last_searched_all, and they are both calculated with `min`

**Comment by mmgoodnow** [comment]
wow this is kind of a huge bugfix. props to you for figuring it out, I spent so much time trying to think about the sql math

**Comment by mmgoodnow** [comment]
would you mind making that rename change and then I'll merge?

File: src/preFilter.ts (hunk 4 above)
Comment (mindset-tk) [reply src/preFilter.ts:80]: Ah, yes, that's a good name change. Let me just push that over.

**Comment by mindset-tk** [comment]
> wow this is kind of a huge bugfix. props to you for figuring it out, I spent so much time trying to think about the sql math

No worries! I work with query langs and still get confused all the time. And Knex syntax added an extra layer of complexity to my brain. Happy to help. I updated that var name.

**Comment by mmgoodnow** [comment]
> Hey, one other thing, I found that leaving line 85 on prefilter.ts as "min" will allow the tool to revisit indexers that were previously filtered by timestamps, see https://github.com/cross-seed/cross-seed/issues/402 - the code for performing the search is already sensitive to if an indexer was recently searched and will filter at that point.

That was the original intention but I guess it never worked.

**Comment by mmgoodnow** [comment]
Thanks for the contribution!

**Pull request merged**